Lib/axisregistry/__init__.py

```diff
@@ -1,52 +1,458 @@
+GF_STATIC_STYLES = OrderedDict(
+    [
+        ("Thin", 100),
+        ("ExtraLight", 200),
+        ("Light", 300),
+        ("Regular", 400),
+        ("Medium", 500),
+        ("SemiBold", 600),
+        ("Bold", 700),
+        ("ExtraBold", 800),
+        ("Black", 900),
+        ("Thin Italic", 100),
+        ("ExtraLight Italic", 200),
+        ("Light Italic", 300),
+        ("Italic", 400),
+        ("Medium Italic", 500),
+        ("SemiBold Italic", 600),
+        ("Bold Italic", 700),
+        ("ExtraBold Italic", 800),
+        ("Black Italic", 900),
+    ]
+)
+
+
+def load_protobuf(klass, path):
+    message = klass()
+    with open(path, "rb") as text_data:
+        text_format.Merge(text_data.read(), message)
+    return message
+
+
+class AxisRegistry:
+    def __init__(self, fp=resource_filename("axisregistry", "data")):
+        axis_fps = [fp for fp in glob(os.path.join(fp, "*.textproto"))]
+        self._data = {}
+        for fp in axis_fps:
+            axis = load_protobuf(AxisProto, fp)
+            self._data[axis.tag] = axis
+
+    def __getitem__(self, k):
+        return self._data[k]
+
+    def __iter__(self):
+        for i in self._data:
+            yield i
+
+    def get_fallback(self, name):
+        for a in self:
+            for fallback in self[a].fallback:
+                if name == fallback.name:
+                    return a, fallback
+        return None, None
+
+    def fallbacks_in_fvar(self, ttFont):
+        res = defaultdict(list)
+        axes_in_font = {
+            a.axisTag: {"min": a.minValue, "max": a.maxValue}
+            for a in ttFont["fvar"].axes
+        }
+        for axis in self:
+            if axis not in axes_in_font:
+                log.warn(f"Axis {axis} not found in GF Axis Registry!")
+                continue
+            for fallback in self[axis].fallback:
+                if (
+                    fallback.value < axes_in_font[axis]["min"]
+                    or fallback.value > axes_in_font[axis]["max"]
+                ):
+                    continue
+                res[axis].append(fallback)
+        return res
+
+    def fallbacks_in_name_table(self, ttFont):
+        res = []
+        name_table = ttFont["name"]
+        tokens = (
+            name_table.getBestFamilyName().split()
+            + name_table.getBestSubFamilyName().split()
+        )
+        fvar_axes_in_font = [a.axisTag for a in ttFont["fvar"].axes]
+        for token in tokens:
+            axis, fallback = axis_registry.get_fallback(token)
+            if not axis or axis in fvar_axes_in_font:
+                continue
+            res.append((axis, fallback))
+        return res
+
+
+axis_registry = AxisRegistry()
+
+
+def is_variable(ttFont):
+    return "fvar" in ttFont
+
+
+def _fvar_dflts(ttFont):
+    res = OrderedDict()
+    for a in ttFont["fvar"].axes:
+        # find name and elision
+        if a.axisTag in axis_registry:
+            name = next(
```

Comment:
I'd make this a method on AxisRegistry: `axis_registry.fallback_name_for_value(a.axisTag, a.defaultValue)`.